src/uct/cuda/cuda_ipc/cuda_ipc_ep.c

```diff
@@ -19,28 +30,313 @@ static UCS_CLASS_INIT_FUNC(uct_cuda_ipc_ep_t, uct_iface_t *tl_iface,
 
     UCS_CLASS_CALL_SUPER_INIT(uct_base_ep_t, &iface->super);
 
+    sglib_hashed_uct_cuda_ipc_rem_seg_t_init(self->rem_segments_hash);
+
     return UCS_OK;
 }
 
 static UCS_CLASS_CLEANUP_FUNC(uct_cuda_ipc_ep_t)
 {
+    uct_cuda_ipc_rem_seg_t *remote_seg;
+    CUresult               cu_ret;
+    struct sglib_hashed_uct_cuda_ipc_rem_seg_t_iterator iter;
+
+    for (remote_seg = sglib_hashed_uct_cuda_ipc_rem_seg_t_it_init(&iter, self->rem_segments_hash);
+         remote_seg != NULL;
+         remote_seg = sglib_hashed_uct_cuda_ipc_rem_seg_t_it_next(&iter)) {
+            sglib_hashed_uct_cuda_ipc_rem_seg_t_delete(self->rem_segments_hash,
+                                                       remote_seg);
+            cu_ret = cuIpcCloseMemHandle(remote_seg->d_bptr);
```

Comment:
I just added driver API wrapper to check return in PR  https://github.com/openucx/ucx/pull/2499/files.
May be we should use macro for all driver calls just to simply the code, for time being you can copy macro to cuda_ipc_iface.h. later we update to base header if that PR goes in

Reply:
Makes sense. I'll replicate this for now.